cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java

```diff
@@ -75,6 +78,66 @@ public Cocos2dxGLSurfaceView getGLSurfaceView(){
+                public void onAudioFocusChange(int focusChange) {
+
+                    Log.d(TAG, "onAudioFocusChange: " + focusChange + ", thread: " + Thread.currentThread().getName());
+
+                    if (mGLSurfaceView == null) {
+                        Log.e(TAG, "mGLSurfaceView is null");
+                        return;
+                    }
+
+                    if (focusChange == AudioManager.AUDIOFOCUS_LOSS) {
+                        // Permanent loss of audio focus
+                        // Pause playback immediately
+                        Log.d(TAG, "Pause music by AUDIOFOCUS_LOSS");
+                        mGLSurfaceView.queueEvent(new Runnable() {
+                            @Override
+                            public void run() {
+                                nativeOnAudioFocusChange(AUDIOFOCUS_LOST);
+                                Cocos2dxHelper.setAudioFocusLost(true);
+                            }
+                        });
+
+                    } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT) {
+                        // Pause playback
+                        Log.d(TAG, "Pause music by AUDIOFOCUS_LOSS_TRANSILENT");
+                        mGLSurfaceView.queueEvent(new Runnable() {
+                            @Override
+                            public void run() {
+                                nativeOnAudioFocusChange(AUDIOFOCUS_LOST_TRANSIENT);
+                                Cocos2dxHelper.setAudioFocusLost(true);
+                            }
+                        });
+                    } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK) {
+                        // Lower the volume, keep playing
+                        Log.d(TAG, "Lower the volume, keep playing by AUDIOFOCUS_LOSS_TRANSILENT_CAN_DUCK");
+                        mGLSurfaceView.queueEvent(new Runnable() {
+                            @Override
+                            public void run() {
+                                nativeOnAudioFocusChange(AUDIOFOCUS_LOST_TRANSIENT_CAN_DUCK);
+                                Cocos2dxHelper.setAudioFocusLost(true);
+                            }
+                        });
+                    } else if (focusChange == AudioManager.AUDIOFOCUS_GAIN) {
+                        // Your app has been granted audio focus again
+                        // Raise volume to normal, restart playback if necessary
+                        Log.d(TAG, "Resume music by AUDIOFOCUS_GAIN");
+                        mGLSurfaceView.queueEvent(new Runnable() {
+                            @Override
+                            public void run() {
+                                nativeOnAudioFocusChange(AUDIOFOCUS_GAIN);
+                                Cocos2dxHelper.setAudioFocusLost(false);
+                            }
+                        });
+                    }
+                }
+            };
+
+
```

Comment:
Can put it into a new java file? Cocos2dxActivity is very complex now, we should put different features into different files.